portfolio/src/main/webapp/script.js

```diff
@@ -159,8 +159,18 @@ function constructCommentNode(comment) {
     timestampNode.classList.add("comment-timestamp");
     timestampNode.innerText = comment.datetime;
 
+    var deleteNode = document.createElement('div');
+    deleteNode.classList.add("comment-delete");
+    var deleteLinkNode = document.createElement('a');
+    deleteLinkNode.classList.add("comment-delete-link");
+    deleteLinkNode.innerText = "×";
+    deleteLinkNode.setAttribute("id", comment.id);
```

Comment:
nit: I would store this attribute as something other than the `id` of the HTML node. To me, the comment's ID in the database would be better suited to identify the container div for that comment on the page, rather than just its delete button.

Alternatively, you can add a custom data attribute, which starts with `data-`: https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/dataset

For example, if you set the property `data-comment-id` on this node, you can access it in JavaScript with `this.dataset.commentId` (note that `comment-id`, the portion after `data-`, becomes `commentId`.

Reply:
oooo, this is super interesting! I'll keep reading up on this and implement changes, thanks!